lsp/server/src/utils/orgUtils.ts

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import {
+    AuthInfo,
+    ConfigAggregator,
+    Connection,
+    OrgConfigProperties,
+    StateAggregator
+} from '@salesforce/core';
+import * as os from 'os';
+import * as fs from 'fs';
+import * as path from 'path';
+import { WorkspaceUtils } from './workspaceUtils';
+import { ObjectInfoRepresentation } from '../types';
+
+enum AuthStatus {
+    UNKNOWN,
+    AUTHORIZED,
+    UNAUTHORIZED
+}
+
+export class OrgUtils {
+    private static orgName: string = '';
+    private static objectInfoFolder = 'objectInfos';
+    private static entityListFileName = 'entity_list.json';
+    private static connection: Connection | undefined;
+
+    private static authStatus: AuthStatus = AuthStatus.UNKNOWN;
+
+    private static objectInfoInMemoCache = new Map<
+        string,
+        ObjectInfoRepresentation
+    >();
+    private static objectInfoPromises = new Map<
+        string,
+        Promise<ObjectInfoRepresentation | undefined>
+    >();
+    private static entities: string[] = [];
+
+    private static sfdxDirWatcher: fs.FSWatcher | undefined;
+    private static sfDirWatcher: fs.FSWatcher | undefined;
+
+    private static sfdxFolder = '.sfdx';
+    /**
+     * The global folder in which sf state is stored.
+     */
+    private static sfFolder = '.sf';
+
+    private static get SFDX_DIR() {
+        return path.join(os.homedir(), this.sfdxFolder);
+    }
+    /**
+     * The full system path to the global sf state folder.
+     */
+    private static get SF_DIR() {
+        return path.join(os.homedir(), this.sfFolder);
+    }
+
+    // Retrieves default organiztion's name.
+    private static async getDefaultOrg(): Promise<string> {
+        const aggregator = await ConfigAggregator.create();
+
+        await aggregator.reload();
+
+        const currentUserConfig = aggregator.getInfo(
+            OrgConfigProperties.TARGET_ORG
+        );
+
+        if (currentUserConfig.value) {
+            this.orgName = currentUserConfig.value.toString();
+            return Promise.resolve(this.orgName);
```

Comment:
You can just return `this.orgName`. It will automatically be wrapped in a promise.